craft_providers/bases/buildd.py

```diff
@@ -755,6 +755,23 @@ def _setup_snapd(
                 capture_output=True,
                 check=True,
             )
+
+            _check_deadline(deadline)
+            http_proxy = self.environment.get("http_proxy")
```

Comment:
if we only setup snapd once; a rerun should update proxy settings, so this might need to live somewhere else

Reply:
I split this code out and added it to `warmup()`